### PR TITLE
Create conversation on compatible target

### PR DESCRIPTION
**Pull request by martinjaimem**
Done:
 - [x] Created conversations for compatible targets
 - [x] Return conversations with the users of compatible targets when creating one
 - [x] Tests for the flow

Related to US:

> As a user I should be able to see all my compatible targets so I can start a conversation with any of my compatible users

### CODE REVIEW

File: app/models/conversation.rb [hunk 1]
```diff
@@ -16,6 +16,24 @@ class Conversation < ApplicationRecord
...
+  scope :users_in_conversation, lambda { |users_id|
+                                  joins('INNER JOIN user_conversations uc_two ON
+                                  conversations.id = uc_two.conversation_id')
+                                    .where('uc_two.user_id': users_id)
+                                }
+
+  def self.get_or_create_by_users(first_user, second_user)
+    old_conversation = first_user.conversations.users_in_conversation([second_user]).first
+
+    return old_conversation if old_conversation
```
Comment (GuilleLeopold) [inline app/models/conversation.rb]: you can join this two lines in:
`old_conversation || Conversation.create!(users: [first_user, second_user])`

File: spec/models/conversation_spec.rb [hunk 2]
```diff
@@ -43,4 +43,19 @@
...
     end
   end
+
+  context 'validate function get_or_create_by_users' do
+    let(:users) { create_list(:user, 3) }
+    it 'creates 2 different conversations with those messages' do
+      expect(Conversation.get_or_create_by_users(users[0], users[1]).users.map(&:id))
+        .to include(users[0][:id], users[1][:id])
+    end
+    context 'conversation of two users was already created' do
```
Comment (GuilleLeopold) [inline spec/models/conversation_spec.rb]: leave an empty line above here

File: spec/models/conversation_spec.rb [hunk 3]
```diff
@@ -43,4 +43,19 @@
...
+      let!(:old_conversation) { create(:conversation, users: [users[0], users[1]]) }
+      it 'returns the conversation already created' do
```
Comment (GuilleLeopold) [inline spec/models/conversation_spec.rb]: same here

File: spec/models/conversation_spec.rb [hunk 4]
```diff
@@ -43,4 +43,19 @@
       is_expected.to include(I18n.t('must be different'))
     end
   end
+
+  context 'validate function get_or_create_by_users' do
+    let(:users) { create_list(:user, 3) }
+    it 'creates 2 different conversations with those messages' do
```
Comment (GuilleLeopold) [inline spec/models/conversation_spec.rb]: leave empty line above

File: spec/controllers/api/v1/targets/targets_create_spec.rb [hunk 5]
```diff
@@ -27,23 +27,34 @@
...
-        I18n.t('api.targets.notification.match'),
-        username: compatible_target.user.username
-      )
+        post api_v1_targets_path, params: params
+        expect(NotificationsService).to have_received(:notify).with(
+          [compatible_target.user.push_token],
+          I18n.t('api.targets.notification.match'),
+          username: compatible_target.user.username
+        )
+        response_body = JSON.parse(response.body)
```
Comment (jdrosales17) [inline spec/controllers/api/v1/targets/targets_create_spec.rb]: I would move this logic to a helper method

File: spec/controllers/api/v1/targets/targets_create_spec.rb [hunk 6]
```diff
@@ -27,23 +27,34 @@
...
-      allow(NotificationsService).to receive(:notify).once.and_call_original
-      params = {
-        target: {
-          title: 'New Target',
-          radius: 3000,
-          topic_id: topic[:id],
-          lat: 45,
-          lng: 45
+    context 'validates notifications and creation of conversations' do
+      it 'notifies the users that matched with the target' do
```
Comment (jdrosales17) [inline spec/controllers/api/v1/targets/targets_create_spec.rb]: This seems too long, can you split it into different test ?

**Pull request merged**